tests/test_functional.py

```diff
@@ -0,0 +1,174 @@
+import asyncio
+
+import pytest
+
+from aviary.core import ToolCall, ToolRequestMessage, fenv
+from aviary.functional import FunctionalEnvironment
+
+
+@pytest.fixture
+def basic_env():
+    @fenv.start()
+    def test_env(param: str):
+        return f"Test {param}", {"param": param}
+
+    @test_env.tool()
+    def test_tool(x: str, state) -> str:  # noqa: ARG001
+        """Test tool that processes input."""
+        return f"Processed {x}"
+
+    return test_env
+
+
+@pytest.fixture
+def async_env():
+    @fenv.start()
+    async def test_env(param: str):
+        await asyncio.sleep(0.1)
+        return f"Async Test {param}", {"param": param}
+
+    @test_env.tool()
+    async def test_tool(x: str, state) -> str:  # noqa: ARG001
+        """Test tool that processes input."""
+        await asyncio.sleep(0.1)
+        return f"Async Processed {x}"
+
+    return test_env
+
+
+def test_environment_creation(basic_env):
+    env = basic_env("test")
+    assert isinstance(env, FunctionalEnvironment)
+    assert len(env.tools) == 1
+    assert env.tools[0].info.name == "test_tool"
+
+
+@pytest.mark.asyncio
+async def test_sync_env_reset(basic_env):
```

Comment:
We call the test `sync_env`, but the fixture `basic_env`. Can you either:
- Rename `basic_env` to `sync_env`
- Rename the test to `test_basic_env_reset`